lib/context_adapter_server.js

```diff
@@ -0,0 +1,427 @@
+                      err.code + ' - ' + err.message,
+                      meta
+                    );
+                    // Notify the opearation as closed
+                    caHelper.notifyOperationClosed(
+                      {
+                        service: service,
+                        servicePath: servicePath,
+                        operationDescriptor: operationDescriptor,
+                        result: caHelper.getResult(err)
+                      },
+                      function() {
+                        // TODO Decide how to behave in case of a notification error
+                      }
+                    );
+                  } else {
+                    // Valid service description retrieved
+                    caLogger.debug(
+                      'service descriptor: ' +
+                      JSON.stringify(serviceDescriptor),
+                      meta
+                    );
+                    // Check the type of action to be accomplished
+                    switch (operationDescriptor.action) {
+                      case caConfig.BUTTON_ENTITY.OPERATION_ACTION.SYNCHRONOUS:
+                        // The requested operation is a synchronous action
+                        // Send a synchronous request to the third party
+                        caHelper.sendThirdPartyRequest(
+                          {
+                            service: request.headers['fiware-service'],
+                            servicePath: request.headers['fiware-servicepath'],
+                            serviceDescriptor: serviceDescriptor,
+                            operationDescriptor: operationDescriptor
+                          },
+                          function(err, response) {
+                            if (err) {
+                              // Error when sending the synchronous request to the third party
+                              caLogger.warn(
+                                'Error when synchronous request from third party: ' +
+                                err.code + ' - ' + err.message,
+                                meta
+                              );
+                              // Notify the oparation as closed
+                              caHelper.notifyOperationClosed(
+                                {
+                                  service: service,
+                                  servicePath: servicePath,
+                                  operationDescriptor: operationDescriptor,
+                                  result: caHelper.getResult(err)
+                                },
+                                function() {
+                                  // TODO Decide how to behave in case of a notification error
+                                }
+                              );
+                            } else {
+                              // The third party responded to the synchronous request successfully
```

Comment:
When you have to put comments to remind you what some parts of the code do, its usually another hint of something that can be extracted to another function.